File: Segundo_Curso/Segundo_Cuatri/PDOO/Estudio/Proyecto_Ruby/Visibilidad/Visibilidad.rb
module Visibilidad
  class Padre
    
    private 
    def privado
	puts "Privado Padre"
    end

    protected
    def protegido
	puts "Protegido Padre"
    end
    
    public 
    def publico
	puts "Publico Padre"
    end

    def test(p)
      privado
      self.privado
      #p.privado
      protegido
      self.protegido
      p.protegido
    end
  end
  class Hija < Padre
    def test(p)
      privado 
      self.privado
#	p.privado
      p.protegido
      protegido
      self.protegido
      publico
      self.publico
      p.publico
    end
  end

  #Padre.new.test(Padre.new)
  #p = Padre.new

  #p.privado
  #p.protegido
  #p.publico
  
  Hija.new.test(Hija.new)
  Hija.new.test(Padre.new)
  h = Hija.new
#  h.privado
#  h.protegido
  h.publico
end
module Externo
    class Hija2 < Visibilidad::Padre	
	protected
	def protegido 
	    puts "Protegido Hija2"
	end

	private
	def privado
	   puts "Privado Hija2"

	end
	
	public
	def test (o)
	    privado
	    o.protegido
	    protegido
	    o.publico
	end
    end
h2 = Hija2.new
h2.test(Hija2.new)
h2.test(Visibilidad::Hija.new)
h2.test(Visibilidad::Padre.new)
end    

File: Segundo_Curso/Segundo_Cuatri/PDOO/Estudio/Proyecto_Java/src/Visibilidad/Visibilidad.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package Visibilidad;
import Visibilidad.otropaquete.HijaOtroPaquete;
/**
 *
 * @author el_dramas
 */
public class Visibilidad {
    private int privado;
    public int publico;
    protected int protegido;
    int paquete;
    
    public void testInstanciaPadre(Visibilidad o){
        System.out.println(o.publico);      //Es publico
        System.out.println(o.privado);      //Estoy en el codigo de la misma clase que o
        System.out.println(o.protegido);    //estoy en el mismo paquete
        System.out.println(o.paquete);      //Estoy en el mismo paquete
    }
    
    public static void testClasePadre(Visibilidad o){
        System.out.println(o.publico);      //Es publico
        System.out.println(o.privado);      //Estoy en el codigo de la misma clase que o
        System.out.println(o.protegido);    //estoy en el mismo paquete
        System.out.println(o.paquete);      //Estoy en el mismo paquete
    }
    
    public void testInstanciaPadre(HijaOtroPaquete o){
        System.out.println(o.publico);      //Es publico
        //System.out.println(o.privado);      //No Estoy en el codigo de la misma clase que o
        System.out.println(o.protegido);    //estoy en el mismo paquete
        //System.out.println(o.paquete);      //No estoy en el mismo paquete
    }
}
